metals/src/main/scala/scala/meta/internal/builds/BuildTools.scala

```diff
@@ -64,9 +64,9 @@ final class BuildTools(
   }
   def isMill: Boolean = workspace.resolve("build.sc").isFile
   def isGradle: Boolean =
-    workspace.resolve("build.gradle").isFile || workspace
-      .resolve("build.gradle.kts")
-      .isFile
+    val defaultGradlePaths = List("settings.gradle", "settings.gradle.kts",
```

Comment:
We'll have to wrap this all in `{}`. However a couple thoughts on this. I think we should for sure add the extra couple root files, but I'm curious if the `./app/build.gradle` and `./app/build.gradle.kts` is necessary. In those situations will there be no gradle file at all in the root? If not, then there is a chance the editor will not be sending in a workspace root to match this. For example in some of the clients we have patterns we look for to detect a workspace root.

https://github.com/scalameta/nvim-metals/blob/main/lua/metals/config.lua#L253

So in the above scenario if there was only a `./app/build.gradle` then the client would send in `./build.gradle` as the root, meaning the `./app/build.gradle` would never be valid _unless_ there was also a build file at the root, and then that's be enough anyways. So a couple things.

1. Some of the clients will need to update their patterns as well probably. I'll do this in `nvim-metals`.
2. I don't think the `./app/build.*` is necessary and actually won't work. wdyt?

Reply:
> meaning the `./app/build.gradle` would never be valid unless there was also a build file at the root, and then that's be enough anyways

I think you're right here

I tried to Google around before sending this small patch and wasn't sure if there was a hard-set Gradle layout for multi-module projects

Just looked again -- I can't find a single example in the Gradle docs or in online comments that doesn't have a `settings.gradle(.kts)` at the project root.

So we should be good with just:

```scala
List("settings.gradle", "settings.gradle.kts", "build.gradle", "build.gradle.kts")
```

Reply:
Fixed in https://github.com/scalameta/metals/pull/3385/commits/64061e52b834008df9f4326e0f299f50818eb4fa 👍